src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1176,7 +1176,13 @@ public function hasCast($key, $types = null)
     public function getCasts()
     {
         if ($this->getIncrementing()) {
-            return array_merge([$this->getKeyName() => $this->getKeyType()], $this->casts);
+            $casts = $this->casts;
+
+            if (! isset($casts[$this->getKeyName()])) {
```

Comment:
This behaviour is actually not the same as before. https://3v4l.org/Q12SM

Reply:
Can you please help me to understand? Probably I'm missing something.

In case of keys conflict, `array_merge()` overrides the values of the first array with the values of `$this->casts`.

Here we check that `$casts` doesn't already have the key set, in that case we set it.

In both scenarios we give the priority to casts, what am I missing?